module/input/Camera/src/settings.h

```diff
@@ -0,0 +1,187 @@
+#ifndef MODULE_INPUT_CAMERA_SETTINGS_H
+#define MODULE_INPUT_CAMERA_SETTINGS_H
+
+extern "C" {
+#include <aravis-0.6/arv.h>
+}
+
+namespace module {
+namespace input {
+    template <typename T>
+    struct SettingsFunctions;
+
+    template <>
+    struct SettingsFunctions<ArvGcInteger> {
+        static int64_t read(ArvGcInteger* setting, GError** error) {
+            return arv_gc_integer_get_value(setting, error);
+        }
+        static void write(ArvGcInteger* setting, const int64_t& v, GError** error) {
+            arv_gc_integer_set_value(setting, v, error);
+        }
+        static bool valid(ArvGcInteger* setting, const int64_t& v) {
+            GError* error = nullptr;
```

Comment:
`std::shared_ptr` with `g_error_free` as deleter?

Reply:
Doesn't work as it's a pointer to a pointer